src/node/rpc/test/membervoting_test.cpp

```diff
@@ -1018,6 +1020,57 @@ TEST_CASE("Complete proposal after initial rejection")
   }
 }
 
+TEST_CASE("Vetoed proposal gets rejected")
+{
+  NetworkTables network;
+  network.tables->set_encryptor(encryptor);
+  Store::Tx gen_tx;
+  GenesisGenerator gen(network, gen_tx);
+  gen.init_values();
+  StubNodeState node;
+  const auto voter_a_cert = get_cert_data(1, kp);
+  auto voter_a = gen.add_member(voter_a_cert, {}, MemberStatus::ACTIVE);
+  const auto voter_b_cert = get_cert_data(2, kp);
+  auto voter_b = gen.add_member(voter_b_cert, {}, MemberStatus::ACTIVE);
+  set_whitelists(gen);
+  gen.set_gov_scripts(lua::Interpreter().invoke<json>(gov_veto_script_file));
+  gen.finalize();
+  MemberRpcFrontend frontend(network, node);
+  frontend.open();
+
+  Script proposal(R"xxx(
+    tables, user_cert = ...
+      return Calls:call("new_user", user_cert)
+    )xxx");
+
+  const vector<uint8_t> user_cert = kp->self_sign("CN=new user");
+  json proposej = create_json_req(Propose::In{proposal, user_cert}, "propose");
+  ccf::SignedReq sr(proposej);
+
+  Response<Propose::Out> r = frontend_process(frontend, proposej, voter_a_cert);
+  CHECK(r.result.completed == false);
+  CHECK(r.result.id == 0);
+
+  const ccf::Script vote_against("return false");
+  {
+    INFO("Member vetoes proposal");
+
+    const auto votej =
+      create_json_req_signed(Vote{0, vote_against}, "vote", kp);
+    Response<bool> r = frontend_process(frontend, votej, voter_b_cert);
+
+    CHECK(r.result == false);
+  }
+
```

Comment:
Would it be worth adding another member `c` to vote for the proposal? So have `a` and `c` (the majority) vote yes and `b` vote no and have the proposal still be rejected.

Reply:
How would that work? If c votes in favour, then the proposal passes and is complete, it cannot be rejected anymore. If b vetoes before c votes in favour, the proposal is rejected and no more votes can be accepted.

Reply:
ah right if we reject the proposal it will not be in `OPEN` state. Still might be worth putting that in the test (i.e. `c` can't vote anymore) but it's up to you :)